dlib/dnn/core.h

```diff
@@ -1430,6 +1436,11 @@ namespace dlib
 
         unsigned int sample_expansion_factor() const { return _sample_expansion_factor; }
 
+        void zero_out_gradients()
```

Comment:
```suggestion
        void set_gradient_inputs_to_zero()
```
How about this name?